File: reposteria_flutter/lib/models/cliente.dart
class Cliente {
  const Cliente({
    required this.id,
    required this.nombre,
    this.telefono,
    this.email,
    this.direccion,
    this.notas,
    required this.activo,
  });
  final int id;
  final String nombre;
  final String? telefono;
  final String? email;
  final String? direccion;
  final String? notas;
  final bool activo;

  factory Cliente.fromJson(Map<String, dynamic> json) => Cliente(
    id: json['id'] as int,
    nombre: json['nombre'] as String,
    telefono: json['telefono'] as String?,
    email: json['email'] as String?,
    direccion: json['direccion'] as String?,
    notas: json['notas'] as String?,
    activo: json['activo'] as bool,
  );
}

File: reposteria-api/app/Models/Cliente.php
<?php

namespace App\Models;

use Database\Factories\ClienteFactory;
use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;

class Cliente extends Model
{
    /** @use HasFactory<ClienteFactory> */
    use HasFactory, SoftDeletes;

    protected $fillable = ['nombre', 'telefono', 'email', 'direccion', 'notas'];

    public function reposteria(): BelongsTo
    {
        return $this->belongsTo(Reposteria::class);
    }

    public function pedidos(): HasMany
    {
        return $this->hasMany(Pedido::class);
    }

    public function ventas(): HasMany
    {
        return $this->hasMany(Venta::class);
    }

    public function scopeDeReposteria(Builder $query, Reposteria|int $reposteria): Builder
    {
        return $query->where('reposteria_id', $reposteria instanceof Reposteria ? $reposteria->id : $reposteria);
    }

    protected function casts(): array
    {
        return ['activo' => 'boolean', 'deleted_at' => 'datetime'];
    }
}
